AzureCalling/External/Authentication/AADAuthHandler.swift

```diff
@@ -53,163 +74,173 @@ class AADAuthHandler {
     }
 
     // MARK: Public API
+    func login(presentingVc: UIViewController,
+               completionHandler: @escaping (Error?) -> Void) {
```

Comment:
Instead of using `Error?` I think we can benefit using `Result`. Success can just be `void` and Failure can provide `Error`

Reply:
Might not even be a bad idea in the Success to return the user information.  That way the AuthHelper doesn't need to save a reference to the user display data (like display name or avatar)

Reply:
We could use Result, but as you've pointed out - the return is void, so I don't see the benefit.